src/kernel/src/namedobj/mod.rs

```diff
@@ -42,6 +45,38 @@ impl NamedObjManager {
 
         Ok(id.into())
     }
+
+    pub fn sys_namedobj_create(
+        self: &Arc<Self>,
+        td: &VThread,
+        i: &SysIn,
+    ) -> Result<SysOut, SysErr> {
+        // Get arguments to pass to namedobj_create
+        let name = unsafe { i.args[0].to_str(32) }?.ok_or(SysErr::Raw(EINVAL))?;
+        let data: usize = i.args[1].into();
+        let flags: u32 = i.args[2].try_into().unwrap();
+
+        self.namedobj_create(td, name, data, flags)
+    }
+
+    fn sys_mdbg_service(self: &Arc<Self>, td: &VThread, i: &SysIn) -> Result<SysOut, SysErr> {
+        // Get the mdbg arg
+        let mdbg_type: i32 = i.args[0].try_into().unwrap();
+
+        match mdbg_type {
+            0x1 => {
```

Comment:
If you look into ghidra, you'll see that this is wrong.
This whole chucnk is missing.
![image](https://github.com/obhq/obliteration/assets/48620541/777c0d7e-0f41-4ace-a11b-5b19a6bcde80)